api/src/functions/cpfValidation/cpfValidation.ts

```diff
@@ -1,28 +1,104 @@
-import { https } from 'https'
-
+import {
+  S3Client,
+  GetObjectCommand,
+  DeleteObjectCommand,
+} from '@aws-sdk/client-s3'
+import { NodeJsClient } from '@smithy/types'
 import { S3Event, S3Handler } from 'aws-lambda'
 
+import { db } from 'src/lib/db'
 import { logger } from 'src/lib/logger'
 
-const apiEndpoint = 'https://example.com'
+const s3 = new S3Client({}) as NodeJsClient<S3Client>
+
+type UploadValidationRecord = {
+  s3: {
+    bucket: {
+      name: string
+    }
+    object: {
+      key: string
+    }
+  }
+}
+
+type UploadValidationS3Client = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  send: (command: GetObjectCommand | DeleteObjectCommand) => Promise<any>
+}
 
-/* eslint-disable @typescript-eslint/no-unused-vars */
 export const handler: S3Handler = async (event: S3Event): Promise<void> => {
-  try {
-    const bucket = event.Records[0].s3.bucket.name
-    const key = event.Records[0].s3.object.key
-
-    const options = {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
+  await Promise.all(
+    event.Records.map(async (record) => {
+      try {
+        await processRecord(record, s3)
+      } catch (err) {
+        logger.error(`Handler error: ${err}`)
+      }
+    })
+  )
+}
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+export const processRecord = async (
+  record: UploadValidationRecord,
+  s3Client: UploadValidationS3Client
+): Promise<void> => {
+  const bucket = record.s3.bucket.name
+  const key = record.s3.object.key
+
+  // Download the JSON errors file from S3
+  const getObjectResponse = await s3Client.send(
+    new GetObjectCommand({ Bucket: bucket, Key: key })
+  )
+
+  // Add the contents of the JSON to the database
+  if (getObjectResponse.Body) {
+    /* example file path/key
+         /uploads/organization_id/agency_id/reporting_period_id/upload_id/{filename}
+      */
+    const result = JSON.parse(getObjectResponse.Body.toString()) || []
+
+    // when the results array is empty then we know the file has passed validations
+    const passed = result.length === 0
+
+    const uploadId = extractUploadIdFromKey(key)
+    const input = {
+      results: result,
+      uploadId: uploadId,
+      passed: passed,
+    }
+    // There should be an existing validation Record in the DB that will need to be updated
+    const validationRecord = await db.uploadValidation.updateMany({
+      data: input,
+      where: { uploadId, passed: false },
```

Comment:
one additional check here will be needed:
```js
where: { uploadId, passed: false, results: { equals: Prisma.JsonNull } }
```

Reply:
Ah good call